app/services/app_integrations/fetch_all.rb

```diff
@@ -0,0 +1,14 @@
+module AppIntegrations
+  class FetchAll
```

Comment:
Suggestion: move this into the `AppIntegrations` class. I'm worried that it's not at all obvious one should call `AppIntegrations::FetchAll.call` instead of `AppIntegration.all` as would be standard with an ActiveRecord model.

Perhaps this check for the supported forem apps could be accomplished with a before hook?

Reply:
@rhymes do you have any thoughts on this? Asking because you suggested this logic could live within a service. 

Even though I understand the worry about not being all that obvious the service should be the way to query for this model, I like the logic being encapsulated in the service object (no need to be concerned about this outside the service object). 

I'm a bit wary about baking this into AR callbacks, but I may be biased and could obviating a good approach to sort this out with the suggested alternative.

Reply:
Hmm, that's a fair point @fdoxyz. My concern is that it's not obvious that one should use this service to get AppIntegrations instead of what's built into the the model via ActiveRecord. Callbacks probably aren't the right choice, but that was the basis of my concern.

I wouldn't consider this comment blocked, I just wanted to put the thought out there!

Reply:
FWIW we also do have a few query objects in `app/queries` and maybe some of these services could live there. I know it adds a little bit to the confusion about where to put things but here's how I think about it:

* `app/models`: "nouns", wrap DB tables. Ideally don't contain too much business logic, e.g. `Thing`
* `app/services`: "verbs", interact with the nouns. Contain most of the business logic, e.g. `Things::DoTheTimewarp.call(params)`
* `app/queries`: noun + "Query`, for more complicated DB retrieval that can't cleanly be expressed as a single AR method, e.g. `Things::MyPreciousQuery.call(params)`

Reply:
This should be theoretically solved by inline documentation. If you group things in a service or query object and, where you use them, explain why you're not using the regular `.all`, people should be able to navigate the code :)

Reply:
My concern was more when writing new code to interact with `AppIntegration` it won't be obvious or discoverable (even with inline documentation in this module class) that one should be using `AppIntegrations::FetchAll` instead of `AppIntegration.all`.

This isn't a blocker for me, and I'm good with whatever you decide @fdoxyz, but I wanted to bring the concern up.

Reply:
Thank you @joshpuetz for sharing the concern and everyone for the input!

TIL about the `app/queries` section of our codebase (hadn't stumbled upon it until now). I think it makes sense to have `AppIntegrations::AllQuery` and `AppIntegrations::FindOrCreateByQuery` (changing the name to more closely relate to AR's `find_or_create_by`).

An idea I have to mitigate the concern would be to override the `.all` method for this class in specific with a call to this custom query and log a warning statement. This way it's not a silent override and someone would hopefully find it and fix it, ideally the developer that added it in the first place. It will continue to work if someone oversees the warning log and doesn't use the Query object (sort of "mistake"), at least avoiding the problem of someone accidentally running the `.all` query without the necessary logic in place. I'll also add some class level comments so this logic is documented as well.

I'm giving a bit more time [to this discussion](https://forem.team/fdoxyz/how-should-we-name-the-admin-dashboard-section-that-holds-mobile-app-configurations-25ah) so I can refactor this PR all in one go. I'll include these changes when I do, in the mean time any other opinions/suggestions are welcome 😄

Reply:
> An idea I have to mitigate the concern would be to override the .all method (only for this class in specific) with a call to this custom query and log a warning statement. 

To be honest, if you go through the trouble of overriding it, I'd probably just call the service from the overridden method instead of creating a warning we might never notice. As long as you return an `ActiveRecord::Relation` this will be pretty much transparent for users of the code. You can still log a warning, but at least an accidental call to the model method will not produce an inconsistent result.

Reply:
"I'll just override/monkeypatch this AR method" - Famous last words 😅 

It became quite difficult to make this work. For now we have the Services refactored into Queries